x/upgrade/keeper/keeper.go

```diff
@@ -74,45 +76,83 @@ func (k Keeper) ScheduleUpgrade(ctx sdk.Context, plan types.Plan) error {
 	}
 
 	store := ctx.KVStore(k.storeKey)
-
 	bz := k.cdc.MustMarshalBinaryBare(&plan)
 	store.Set(types.PlanKey(), bz)
 
-	if plan.UpgradedClientState != nil {
-		clientState, err := clienttypes.UnpackClientState(plan.UpgradedClientState)
-		if err != nil {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "could not unpack clientstate: %v", err)
-		}
-		return k.SetUpgradedClient(ctx, clientState)
+	if plan.UpgradedClientState == nil {
+		// if latest UpgradedClientState is nil, but upgraded client exists in store,
+		// then delete client state from store.
+		_, height, _ := k.GetUpgradedClient(ctx)
```

Comment:
I think you need to check if `err == nil` or `height != 0` before deleting? otherwise I'm not sure why the check is used in `SetUpgradeHeight`

Reply:
I think delete performs a no-op if the key is not found. But I agree with the comment, we should be checking for err == nil

Reply:
Delete is a no-op yes, I have to do this here as well, because `SetUpgradedClient` isn't called here but I still want to delete the upgraded client